backend/company_api.py

```
@@ -154,6 +200,48 @@ async def get_optional_company_access(
 # COMPANY ENDPOINTS
 # =============================================================================
 
+@router.get("", response_model=dict)
+async def list_companies(
+    limit: int = Query(100, ge=1, le=1000),
+    offset: int = Query(0, ge=0),
+    search: str | None = Query(None, description="Search by name or domain"),
+    user: dict = Depends(_get_current_user_thunk),
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)